### ISSUE TITLE
Project 'customModules' not found in project ':server'.

### ISSUE THREAD

**Issue by seanthingee** [PopMedNet-Team/Mobile-Application#24]
I'm at step 3.2.1 of the [documentation](https://github.com/PopMedNet-Team/FDA-My-Studies-Mobile-Application-System/blob/2019.05/FDAMyStudies_Release%202019.05__SetupInstructions_V1.0.pdf) and I am getting the following error:

```
$ ./gradlew -PdeployMode=prod :server:customModules:fdahpUserRegWS:distributions:Registration:distribution

<-------------> 0% INITIALIZING [0s]

FAILURE: Build failed with an exception.

* What went wrong:
Project 'customModules' not found in project ':server'.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 6.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/5.2.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 3s
```

After fiddling around, I added `include ":server:customModules:fdahpUserRegWS"` too settings.gradle. Now I get a more interesting error:

```
$  ./gradlew -PdeployMode=prod :server:customModules:fdahpUserRegWS:distributions:Registration:distribution

FAILURE: Build failed with an exception.

* What went wrong:
Project 'distributions' not found in project ':server:customModules:fdahpUserRegWS'.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 6.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/5.2.1/userguide/command_line_interface.html#sec:command_line_warnings
```

The fact that distributions cannot be found kind of makes sense since there isn't anything (files/folders) in the repository named "distributions". Nor "Registration". 

Did anyone else manage to get this step to work?

**Comment by elliot2extreme** [comment]
Please try the following to get it working:

1. Rename the folder UserReg-WS to fdahpUserRegWS
2. Add `include ":server:customModules:fdahpUserRegWS"` to `settings.gradle` in `{LABKEY_HOME}`
3. In terminal switch to folder `{LABKEY_HOME}/server/customModules/fdahpUserRegWS`
4. Execute `{LABKEY_HOME}/gradlew deployModule -PdeployMode=prod`
5. Now the module is working in dev environment

Next I have to figure out how to deploy it to another Tomcat instance.

**Comment by seanthingee** [comment]
@elliot2extreme 
Really appreciate you commenting but it's not working.

```
[centos@ip-10-57-0-123 fdahpUserRegWS]$ ~/labkey/trunk/gradlew deployModule -PdeployMode=prod

> Task :server:customModules:fdahpUserRegWS:compileJava FAILED
/home/centos/labkey/trunk/server/customModules/fdahpUserRegWS/src/com/hphc/mystudies/FdahpUserRegWSSchema.java:71: error: method getVersionInfo() is already defined in class FdahpUserRegWSSchema
    public TableInfo getVersionInfo(){return getSchema().getTable("VersionInfo");}
                     ^
Note: /home/centos/labkey/trunk/server/customModules/fdahpUserRegWS/src/com/hphc/mystudies/FdahpUserRegWSController.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: /home/centos/labkey/trunk/server/customModules/fdahpUserRegWS/src/com/hphc/mystudies/FdaAuditProvider.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
1 error

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':server:customModules:fdahpUserRegWS:compileJava'.
> Compilation failed; see the compiler error output for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 6.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/5.2.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 4s
22 actionable tasks: 2 executed, 20 up-to-date

```

**Comment by elliot2extreme** [comment]
Sorry, i forgot this point. There is exactly the same function `getVersionInfo()` too times in the Java class `fdahpUserRegWS/src/com/hphc/mystudies/FdahpUserRegWSSchema.java` and you have to delete one. Afterwards its working.

**Comment by seanthingee** [comment]
@elliot2extreme Thank you, I got it to build.

**Comment by aaronboss** [comment]
I had the same issue that @seanthingee described at the top. I tried to build inside of server/customModules/fdahpUserRegWS but received this:

```
FAILURE: Build failed with an exception.

* What went wrong:
Task 'deployModule' not found in project ':server:customModules:fdahpUserRegWS'.

```

Is there something I'm missing?

**Comment by seanthingee** [comment]
@aaronboss I don't know what is happening, but maybe you could post the commands that lead you to this point, just so they can be confirmed.

Did you follow the instructions in section 3.1 in the Setup Instructions PDF where it says "Switch to the release 19.1 branch and then do a git pull"? It might have nothing to do with what you are dealing with, but I can imagine your issue could be related to an incompatibility with the version of Labkey. Btw, their instruction is incorrect: you need to change branches in the _Labkey_ repo, which is _SVN_. 

```
% cd $LABKEY_HOME
% svn switch https://svn.mgt.labkey.host/stedi/branches/release19.1
```

**Comment by aaronboss** [comment]
@seanthingee  Thank you very much with the switch. I didn't realize I wasn't on 19.1, but on the trunk.

This still seems to be broken for me.

So the instructions I followed were on
https://www.labkey.org/Documentation/wiki-page.view?name=devMachine

After I finished those instructions successfully, I went back to the Setup Instructions from the MyStudies documentation.

I copied the UserReg-WS directory to customModules inside of labkey. I renamed that directory to fdahpUserRegWS.

I ran the steps in 3.2.1, all worked fine except for the :server:customModules:fdahpUserRegWS:distributions:Registration:distribution command.

This gave me the exception 

```
 FAILURE: Build failed with an exception.

* What went wrong:
Project 'distributions' not found in project ':server:customModules:fdahpUserRegWS'.
```
I followed the directions from @elliot2extreme with building inside of the fdahpUserRegWS directory, but still receive 

```
:~/development/labkey/trunk/server/customModules/fdahpUserRegWS$ $LABKEY_HOME/gradlew deployModule

FAILURE: Build failed with an exception.

* What went wrong:
Task 'deployModule' not found in project ':server:customModules:fdahpUserRegWS'.

* Try:
Run gradlew tasks to get a list of available tasks. Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 6.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/5.2.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 3s

```

**Comment by seifscape** [comment]
> @seanthingee Thank you very much with the switch. I didn't realize I wasn't on 19.1, but on the trunk.
> 
> This still seems to be broken for me.
> 
> So the instructions I followed were on
> https://www.labkey.org/Documentation/wiki-page.view?name=devMachine
> 
> After I finished those instructions successfully, I went back to the Setup Instructions from the MyStudies documentation.
> 
> I copied the UserReg-WS directory to customModules inside of labkey. I renamed that directory to fdahpUserRegWS.
> 
> I ran the steps in 3.2.1, all worked fine except for the :server:customModules:fdahpUserRegWS:distributions:Registration:distribution command.
> 
> This gave me the exception
> 
> ```
>  FAILURE: Build failed with an exception.
> 
> * What went wrong:
> Project 'distributions' not found in project ':server:customModules:fdahpUserRegWS'.
> ```
> 
> I followed the directions from @elliot2extreme with building inside of the fdahpUserRegWS directory, but still receive
> 
> ```
> :~/development/labkey/trunk/server/customModules/fdahpUserRegWS$ $LABKEY_HOME/gradlew deployModule
> 
> FAILURE: Build failed with an exception.
> 
> * What went wrong:
> Task 'deployModule' not found in project ':server:customModules:fdahpUserRegWS'.
> 
> * Try:
> Run gradlew tasks to get a list of available tasks. Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
> 
> * Get more help at https://help.gradle.org
> 
> Deprecated Gradle features were used in this build, making it incompatible with Gradle 6.0.
> Use '--warning-mode all' to show the individual deprecation warnings.
> See https://docs.gradle.org/5.2.1/userguide/command_line_interface.html#sec:command_line_warnings
> 
> BUILD FAILED in 3s
> ```

@aaronboss  Were you able to get past this?

**Comment by FDA-MyStudies** [comment]
Hi @seanthingee ,

Thanks for your interest in the MyStudies system. Apologies for the delay in response.  This issue has been addressed in 2019.10 release by adding a distribution folder under UserReg-WS project. Download the distributions folder inside UserReg-WS from the Github and place it inside your project and then try running the command. Also our document is updated with more details with folder naming, refer the latest instruction document section 3.2.1

https://github.com/PopMedNet-Team/FDA-My-Studies-Mobile-Application-System/blob/2019.10/FDAMyStudies_Release%202019.10_SetupInstructions_V1.0.pdf"
